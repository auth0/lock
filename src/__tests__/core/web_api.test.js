import Auth0WebApi from '../../core/web_api';

describe('Auth0WebApi', () => {
  let originalWindow;

  const LOCK_ID = 'lock-id';
  const CLIENT_ID = 'client-id';
  const DEFAULT_DOMAIN = 'test.com';
  const client = () => Auth0WebApi.clients[LOCK_ID];

  beforeEach(() => {
    originalWindow = global.window;
  });

  afterEach(() => {
    global.window = originalWindow;
  });

  describe('setupClient', () => {
    it('sets the correct options when is on the hosted login page', () => {
      delete global.window.location;
      window.location = { ...originalWindow.location, host: DEFAULT_DOMAIN, search: '' };
      Auth0WebApi.setupClient(LOCK_ID, CLIENT_ID, DEFAULT_DOMAIN, { redirect: true });

      expect(client()).toEqual(
        expect.objectContaining({
          isUniversalLogin: true,
          domain: DEFAULT_DOMAIN,
          authOpt: {
            popup: false
          }
        })
      );
    });

    it('sets redirect: true when on the same origin as the specified domain', () => {
      delete global.window.location;
      window.location = { ...originalWindow.location, host: DEFAULT_DOMAIN, search: '' };

      Auth0WebApi.setupClient(LOCK_ID, CLIENT_ID, DEFAULT_DOMAIN, {});
      expect(client().authOpt.popup).toBe(false);
    });

    it('sets redirect: false when on a different origin as the specified domain', () => {
      delete global.window.location;
      window.location = { ...originalWindow.location, host: 'test-other.com', search: '' };

      Auth0WebApi.setupClient(LOCK_ID, CLIENT_ID, DEFAULT_DOMAIN, {});
      expect(client().authOpt.popup).toBe(true);
    });

    it('forces popup and sso mode for cordova, only when not running in the hosted environment', () => {
      delete global.window.location;
      window.location = { ...originalWindow.location, host: DEFAULT_DOMAIN, search: '' };
      window.cordova = true;

      Auth0WebApi.setupClient(LOCK_ID, CLIENT_ID, DEFAULT_DOMAIN, {});
      expect(client().authOpt.popup).toBe(false);
      expect(client().authOpt.sso).toBeUndefined();
    });

    it('forces popup and sso mode for electron, only when not running in the hosted environment', () => {
      delete global.window.location;
      window.location = { ...originalWindow.location, host: DEFAULT_DOMAIN, search: '' };
      window.electron = true;

      Auth0WebApi.setupClient(LOCK_ID, CLIENT_ID, DEFAULT_DOMAIN, {});
      expect(client().authOpt.popup).toBe(false);
      expect(client().authOpt.sso).toBeUndefined();
    });
  });
});
