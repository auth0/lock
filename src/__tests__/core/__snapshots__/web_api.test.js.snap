// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Auth0WebApi setupClient sets the correct options when is on the hosted login page 1`] = `
Auth0WebAPI {
  "clients": Object {
    "lock-id": Auth0APIClient {
      "_enableIdPInitiatedLogin": false,
      "authOpt": Object {
        "nonce": undefined,
        "popup": false,
        "popupOptions": undefined,
        "scope": undefined,
        "state": undefined,
      },
      "client": WebAuth {
        "_universalLogin": HostedPages {
          "baseOptions": Object {
            "_sendTelemetry": true,
            "_telemetryInfo": Object {
              "env": Object {
                "auth0.js": "9.13.2",
              },
              "name": "lock.js",
              "version": "11.24.1",
            },
            "_timesToRetryFailedRequests": 0,
            "audience": undefined,
            "clientID": "client-id",
            "domain": "test.com",
            "jwksURI": null,
            "leeway": 60,
            "nonce": undefined,
            "overrides": null,
            "plugins": PluginHandler {
              "plugins": Array [
                CordovaPlugin {
                  "extensibilityPoints": Array [
                    "popup.authorize",
                    "popup.getPopupHandler",
                  ],
                  "version": "9.13.2",
                  "webAuth": [Circular],
                },
              ],
            },
            "popupOrigin": undefined,
            "redirectUri": undefined,
            "responseMode": undefined,
            "responseType": undefined,
            "rootUrl": "https://test.com",
            "scope": undefined,
            "state": undefined,
            "tenant": "test",
            "token_issuer": "https://test.com/",
            "universalLoginPage": true,
          },
          "client": [Circular],
          "request": RequestBuilder {
            "_sendTelemetry": true,
            "_telemetryInfo": Object {
              "env": Object {
                "auth0.js": "9.13.2",
              },
              "name": "lock.js",
              "version": "11.24.1",
            },
            "_timesToRetryFailedRequests": 0,
            "_universalLoginPage": true,
            "headers": Object {},
          },
          "warn": Warn {
            "disableWarnings": false,
          },
        },
        "baseOptions": Object {
          "_sendTelemetry": true,
          "_telemetryInfo": Object {
            "env": Object {
              "auth0.js": "9.13.2",
            },
            "name": "lock.js",
            "version": "11.24.1",
          },
          "_timesToRetryFailedRequests": 0,
          "audience": undefined,
          "clientID": "client-id",
          "domain": "test.com",
          "jwksURI": null,
          "leeway": 60,
          "nonce": undefined,
          "overrides": null,
          "plugins": PluginHandler {
            "plugins": Array [
              CordovaPlugin {
                "extensibilityPoints": Array [
                  "popup.authorize",
                  "popup.getPopupHandler",
                ],
                "version": "9.13.2",
                "webAuth": [Circular],
              },
            ],
          },
          "popupOrigin": undefined,
          "redirectUri": undefined,
          "responseMode": undefined,
          "responseType": undefined,
          "rootUrl": "https://test.com",
          "scope": undefined,
          "state": undefined,
          "tenant": "test",
          "token_issuer": "https://test.com/",
          "universalLoginPage": true,
        },
        "client": Authentication {
          "baseOptions": Object {
            "_sendTelemetry": true,
            "_telemetryInfo": Object {
              "env": Object {
                "auth0.js": "9.13.2",
              },
              "name": "lock.js",
              "version": "11.24.1",
            },
            "_timesToRetryFailedRequests": 0,
            "audience": undefined,
            "clientID": "client-id",
            "domain": "test.com",
            "jwksURI": null,
            "leeway": 60,
            "nonce": undefined,
            "overrides": null,
            "plugins": PluginHandler {
              "plugins": Array [
                CordovaPlugin {
                  "extensibilityPoints": Array [
                    "popup.authorize",
                    "popup.getPopupHandler",
                  ],
                  "version": "9.13.2",
                  "webAuth": [Circular],
                },
              ],
            },
            "popupOrigin": undefined,
            "redirectUri": undefined,
            "responseMode": undefined,
            "responseType": undefined,
            "rootUrl": "https://test.com",
            "scope": undefined,
            "state": undefined,
            "tenant": "test",
            "token_issuer": "https://test.com/",
            "universalLoginPage": true,
          },
          "dbConnection": DBConnection {
            "baseOptions": Object {
              "_sendTelemetry": true,
              "_telemetryInfo": Object {
                "env": Object {
                  "auth0.js": "9.13.2",
                },
                "name": "lock.js",
                "version": "11.24.1",
              },
              "_timesToRetryFailedRequests": 0,
              "audience": undefined,
              "clientID": "client-id",
              "domain": "test.com",
              "jwksURI": null,
              "leeway": 60,
              "nonce": undefined,
              "overrides": null,
              "plugins": PluginHandler {
                "plugins": Array [
                  CordovaPlugin {
                    "extensibilityPoints": Array [
                      "popup.authorize",
                      "popup.getPopupHandler",
                    ],
                    "version": "9.13.2",
                    "webAuth": [Circular],
                  },
                ],
              },
              "popupOrigin": undefined,
              "redirectUri": undefined,
              "responseMode": undefined,
              "responseType": undefined,
              "rootUrl": "https://test.com",
              "scope": undefined,
              "state": undefined,
              "tenant": "test",
              "token_issuer": "https://test.com/",
              "universalLoginPage": true,
            },
            "request": RequestBuilder {
              "_sendTelemetry": true,
              "_telemetryInfo": Object {
                "env": Object {
                  "auth0.js": "9.13.2",
                },
                "name": "lock.js",
                "version": "11.24.1",
              },
              "_timesToRetryFailedRequests": 0,
              "_universalLoginPage": undefined,
              "headers": Object {},
            },
          },
          "passwordless": PasswordlessAuthentication {
            "baseOptions": Object {
              "_sendTelemetry": true,
              "_telemetryInfo": Object {
                "env": Object {
                  "auth0.js": "9.13.2",
                },
                "name": "lock.js",
                "version": "11.24.1",
              },
              "_timesToRetryFailedRequests": 0,
              "audience": undefined,
              "clientID": "client-id",
              "domain": "test.com",
              "jwksURI": null,
              "leeway": 60,
              "nonce": undefined,
              "overrides": null,
              "plugins": PluginHandler {
                "plugins": Array [
                  CordovaPlugin {
                    "extensibilityPoints": Array [
                      "popup.authorize",
                      "popup.getPopupHandler",
                    ],
                    "version": "9.13.2",
                    "webAuth": [Circular],
                  },
                ],
              },
              "popupOrigin": undefined,
              "redirectUri": undefined,
              "responseMode": undefined,
              "responseType": undefined,
              "rootUrl": "https://test.com",
              "scope": undefined,
              "state": undefined,
              "tenant": "test",
              "token_issuer": "https://test.com/",
              "universalLoginPage": true,
            },
            "request": RequestBuilder {
              "_sendTelemetry": true,
              "_telemetryInfo": Object {
                "env": Object {
                  "auth0.js": "9.13.2",
                },
                "name": "lock.js",
                "version": "11.24.1",
              },
              "_timesToRetryFailedRequests": 0,
              "_universalLoginPage": undefined,
              "headers": Object {},
            },
          },
          "request": RequestBuilder {
            "_sendTelemetry": true,
            "_telemetryInfo": Object {
              "env": Object {
                "auth0.js": "9.13.2",
              },
              "name": "lock.js",
              "version": "11.24.1",
            },
            "_timesToRetryFailedRequests": 0,
            "_universalLoginPage": undefined,
            "headers": Object {},
          },
          "ssodataStorage": SSODataStorage {
            "storage": Storage {
              "handler": StorageHandler {
                "storage": CookieStorage {},
                "warn": Warn {
                  "disableWarnings": undefined,
                },
              },
            },
          },
          "warn": Warn {
            "disableWarnings": false,
          },
        },
        "crossOriginAuthentication": CrossOriginAuthentication {
          "baseOptions": Object {
            "_sendTelemetry": true,
            "_telemetryInfo": Object {
              "env": Object {
                "auth0.js": "9.13.2",
              },
              "name": "lock.js",
              "version": "11.24.1",
            },
            "_timesToRetryFailedRequests": 0,
            "audience": undefined,
            "clientID": "client-id",
            "domain": "test.com",
            "jwksURI": null,
            "leeway": 60,
            "nonce": undefined,
            "overrides": null,
            "plugins": PluginHandler {
              "plugins": Array [
                CordovaPlugin {
                  "extensibilityPoints": Array [
                    "popup.authorize",
                    "popup.getPopupHandler",
                  ],
                  "version": "9.13.2",
                  "webAuth": [Circular],
                },
              ],
            },
            "popupOrigin": undefined,
            "redirectUri": undefined,
            "responseMode": undefined,
            "responseType": undefined,
            "rootUrl": "https://test.com",
            "scope": undefined,
            "state": undefined,
            "tenant": "test",
            "token_issuer": "https://test.com/",
            "universalLoginPage": true,
          },
          "request": RequestBuilder {
            "_sendTelemetry": true,
            "_telemetryInfo": Object {
              "env": Object {
                "auth0.js": "9.13.2",
              },
              "name": "lock.js",
              "version": "11.24.1",
            },
            "_timesToRetryFailedRequests": 0,
            "_universalLoginPage": undefined,
            "headers": Object {},
          },
          "storage": Storage {
            "handler": StorageHandler {
              "storage": CookieStorage {},
              "warn": Warn {
                "disableWarnings": undefined,
              },
            },
          },
          "webAuth": [Circular],
          "webMessageHandler": WebMessageHandler {
            "warn": Warn {
              "disableWarnings": undefined,
            },
            "webAuth": [Circular],
          },
        },
        "popup": Popup {
          "baseOptions": Object {
            "_sendTelemetry": true,
            "_telemetryInfo": Object {
              "env": Object {
                "auth0.js": "9.13.2",
              },
              "name": "lock.js",
              "version": "11.24.1",
            },
            "_timesToRetryFailedRequests": 0,
            "audience": undefined,
            "clientID": "client-id",
            "domain": "test.com",
            "jwksURI": null,
            "leeway": 60,
            "nonce": undefined,
            "overrides": null,
            "plugins": PluginHandler {
              "plugins": Array [
                CordovaPlugin {
                  "extensibilityPoints": Array [
                    "popup.authorize",
                    "popup.getPopupHandler",
                  ],
                  "version": "9.13.2",
                  "webAuth": [Circular],
                },
              ],
            },
            "popupOrigin": undefined,
            "redirectUri": undefined,
            "responseMode": undefined,
            "responseType": undefined,
            "rootUrl": "https://test.com",
            "scope": undefined,
            "state": undefined,
            "tenant": "test",
            "token_issuer": "https://test.com/",
            "universalLoginPage": true,
          },
          "client": Authentication {
            "baseOptions": Object {
              "_sendTelemetry": true,
              "_telemetryInfo": Object {
                "env": Object {
                  "auth0.js": "9.13.2",
                },
                "name": "lock.js",
                "version": "11.24.1",
              },
              "_timesToRetryFailedRequests": 0,
              "audience": undefined,
              "clientID": "client-id",
              "domain": "test.com",
              "jwksURI": null,
              "leeway": 60,
              "nonce": undefined,
              "overrides": null,
              "plugins": PluginHandler {
                "plugins": Array [
                  CordovaPlugin {
                    "extensibilityPoints": Array [
                      "popup.authorize",
                      "popup.getPopupHandler",
                    ],
                    "version": "9.13.2",
                    "webAuth": [Circular],
                  },
                ],
              },
              "popupOrigin": undefined,
              "redirectUri": undefined,
              "responseMode": undefined,
              "responseType": undefined,
              "rootUrl": "https://test.com",
              "scope": undefined,
              "state": undefined,
              "tenant": "test",
              "token_issuer": "https://test.com/",
              "universalLoginPage": true,
            },
            "dbConnection": DBConnection {
              "baseOptions": Object {
                "_sendTelemetry": true,
                "_telemetryInfo": Object {
                  "env": Object {
                    "auth0.js": "9.13.2",
                  },
                  "name": "lock.js",
                  "version": "11.24.1",
                },
                "_timesToRetryFailedRequests": 0,
                "audience": undefined,
                "clientID": "client-id",
                "domain": "test.com",
                "jwksURI": null,
                "leeway": 60,
                "nonce": undefined,
                "overrides": null,
                "plugins": PluginHandler {
                  "plugins": Array [
                    CordovaPlugin {
                      "extensibilityPoints": Array [
                        "popup.authorize",
                        "popup.getPopupHandler",
                      ],
                      "version": "9.13.2",
                      "webAuth": [Circular],
                    },
                  ],
                },
                "popupOrigin": undefined,
                "redirectUri": undefined,
                "responseMode": undefined,
                "responseType": undefined,
                "rootUrl": "https://test.com",
                "scope": undefined,
                "state": undefined,
                "tenant": "test",
                "token_issuer": "https://test.com/",
                "universalLoginPage": true,
              },
              "request": RequestBuilder {
                "_sendTelemetry": true,
                "_telemetryInfo": Object {
                  "env": Object {
                    "auth0.js": "9.13.2",
                  },
                  "name": "lock.js",
                  "version": "11.24.1",
                },
                "_timesToRetryFailedRequests": 0,
                "_universalLoginPage": undefined,
                "headers": Object {},
              },
            },
            "passwordless": PasswordlessAuthentication {
              "baseOptions": Object {
                "_sendTelemetry": true,
                "_telemetryInfo": Object {
                  "env": Object {
                    "auth0.js": "9.13.2",
                  },
                  "name": "lock.js",
                  "version": "11.24.1",
                },
                "_timesToRetryFailedRequests": 0,
                "audience": undefined,
                "clientID": "client-id",
                "domain": "test.com",
                "jwksURI": null,
                "leeway": 60,
                "nonce": undefined,
                "overrides": null,
                "plugins": PluginHandler {
                  "plugins": Array [
                    CordovaPlugin {
                      "extensibilityPoints": Array [
                        "popup.authorize",
                        "popup.getPopupHandler",
                      ],
                      "version": "9.13.2",
                      "webAuth": [Circular],
                    },
                  ],
                },
                "popupOrigin": undefined,
                "redirectUri": undefined,
                "responseMode": undefined,
                "responseType": undefined,
                "rootUrl": "https://test.com",
                "scope": undefined,
                "state": undefined,
                "tenant": "test",
                "token_issuer": "https://test.com/",
                "universalLoginPage": true,
              },
              "request": RequestBuilder {
                "_sendTelemetry": true,
                "_telemetryInfo": Object {
                  "env": Object {
                    "auth0.js": "9.13.2",
                  },
                  "name": "lock.js",
                  "version": "11.24.1",
                },
                "_timesToRetryFailedRequests": 0,
                "_universalLoginPage": undefined,
                "headers": Object {},
              },
            },
            "request": RequestBuilder {
              "_sendTelemetry": true,
              "_telemetryInfo": Object {
                "env": Object {
                  "auth0.js": "9.13.2",
                },
                "name": "lock.js",
                "version": "11.24.1",
              },
              "_timesToRetryFailedRequests": 0,
              "_universalLoginPage": undefined,
              "headers": Object {},
            },
            "ssodataStorage": SSODataStorage {
              "storage": Storage {
                "handler": StorageHandler {
                  "storage": CookieStorage {},
                  "warn": Warn {
                    "disableWarnings": undefined,
                  },
                },
              },
            },
            "warn": Warn {
              "disableWarnings": false,
            },
          },
          "crossOriginAuthentication": CrossOriginAuthentication {
            "baseOptions": Object {
              "_sendTelemetry": true,
              "_telemetryInfo": Object {
                "env": Object {
                  "auth0.js": "9.13.2",
                },
                "name": "lock.js",
                "version": "11.24.1",
              },
              "_timesToRetryFailedRequests": 0,
              "audience": undefined,
              "clientID": "client-id",
              "domain": "test.com",
              "jwksURI": null,
              "leeway": 60,
              "nonce": undefined,
              "overrides": null,
              "plugins": PluginHandler {
                "plugins": Array [
                  CordovaPlugin {
                    "extensibilityPoints": Array [
                      "popup.authorize",
                      "popup.getPopupHandler",
                    ],
                    "version": "9.13.2",
                    "webAuth": [Circular],
                  },
                ],
              },
              "popupOrigin": undefined,
              "redirectUri": undefined,
              "responseMode": undefined,
              "responseType": undefined,
              "rootUrl": "https://test.com",
              "scope": undefined,
              "state": undefined,
              "tenant": "test",
              "token_issuer": "https://test.com/",
              "universalLoginPage": true,
            },
            "request": RequestBuilder {
              "_sendTelemetry": true,
              "_telemetryInfo": Object {
                "env": Object {
                  "auth0.js": "9.13.2",
                },
                "name": "lock.js",
                "version": "11.24.1",
              },
              "_timesToRetryFailedRequests": 0,
              "_universalLoginPage": undefined,
              "headers": Object {},
            },
            "storage": Storage {
              "handler": StorageHandler {
                "storage": CookieStorage {},
                "warn": Warn {
                  "disableWarnings": undefined,
                },
              },
            },
            "webAuth": [Circular],
            "webMessageHandler": WebMessageHandler {
              "warn": Warn {
                "disableWarnings": undefined,
              },
              "webAuth": [Circular],
            },
          },
          "transactionManager": TransactionManager {
            "keyLength": 32,
            "namespace": "com.auth0.auth.",
            "options": Object {
              "_sendTelemetry": true,
              "_telemetryInfo": Object {
                "env": Object {
                  "auth0.js": "9.13.2",
                },
                "name": "lock.js",
                "version": "11.24.1",
              },
              "_timesToRetryFailedRequests": 0,
              "audience": undefined,
              "clientID": "client-id",
              "domain": "test.com",
              "jwksURI": null,
              "leeway": 60,
              "nonce": undefined,
              "overrides": null,
              "plugins": PluginHandler {
                "plugins": Array [
                  CordovaPlugin {
                    "extensibilityPoints": Array [
                      "popup.authorize",
                      "popup.getPopupHandler",
                    ],
                    "version": "9.13.2",
                    "webAuth": [Circular],
                  },
                ],
              },
              "popupOrigin": undefined,
              "redirectUri": undefined,
              "responseMode": undefined,
              "responseType": undefined,
              "rootUrl": "https://test.com",
              "scope": undefined,
              "state": undefined,
              "tenant": "test",
              "token_issuer": "https://test.com/",
              "universalLoginPage": true,
            },
            "storage": Storage {
              "handler": StorageHandler {
                "storage": CookieStorage {},
                "warn": Warn {
                  "disableWarnings": undefined,
                },
              },
            },
          },
          "warn": Warn {
            "disableWarnings": false,
          },
          "webAuth": [Circular],
        },
        "redirect": Redirect {
          "baseOptions": Object {
            "_sendTelemetry": true,
            "_telemetryInfo": Object {
              "env": Object {
                "auth0.js": "9.13.2",
              },
              "name": "lock.js",
              "version": "11.24.1",
            },
            "_timesToRetryFailedRequests": 0,
            "audience": undefined,
            "clientID": "client-id",
            "domain": "test.com",
            "jwksURI": null,
            "leeway": 60,
            "nonce": undefined,
            "overrides": null,
            "plugins": PluginHandler {
              "plugins": Array [
                CordovaPlugin {
                  "extensibilityPoints": Array [
                    "popup.authorize",
                    "popup.getPopupHandler",
                  ],
                  "version": "9.13.2",
                  "webAuth": [Circular],
                },
              ],
            },
            "popupOrigin": undefined,
            "redirectUri": undefined,
            "responseMode": undefined,
            "responseType": undefined,
            "rootUrl": "https://test.com",
            "scope": undefined,
            "state": undefined,
            "tenant": "test",
            "token_issuer": "https://test.com/",
            "universalLoginPage": true,
          },
          "crossOriginAuthentication": CrossOriginAuthentication {
            "baseOptions": Object {
              "_sendTelemetry": true,
              "_telemetryInfo": Object {
                "env": Object {
                  "auth0.js": "9.13.2",
                },
                "name": "lock.js",
                "version": "11.24.1",
              },
              "_timesToRetryFailedRequests": 0,
              "audience": undefined,
              "clientID": "client-id",
              "domain": "test.com",
              "jwksURI": null,
              "leeway": 60,
              "nonce": undefined,
              "overrides": null,
              "plugins": PluginHandler {
                "plugins": Array [
                  CordovaPlugin {
                    "extensibilityPoints": Array [
                      "popup.authorize",
                      "popup.getPopupHandler",
                    ],
                    "version": "9.13.2",
                    "webAuth": [Circular],
                  },
                ],
              },
              "popupOrigin": undefined,
              "redirectUri": undefined,
              "responseMode": undefined,
              "responseType": undefined,
              "rootUrl": "https://test.com",
              "scope": undefined,
              "state": undefined,
              "tenant": "test",
              "token_issuer": "https://test.com/",
              "universalLoginPage": true,
            },
            "request": RequestBuilder {
              "_sendTelemetry": true,
              "_telemetryInfo": Object {
                "env": Object {
                  "auth0.js": "9.13.2",
                },
                "name": "lock.js",
                "version": "11.24.1",
              },
              "_timesToRetryFailedRequests": 0,
              "_universalLoginPage": undefined,
              "headers": Object {},
            },
            "storage": Storage {
              "handler": StorageHandler {
                "storage": CookieStorage {},
                "warn": Warn {
                  "disableWarnings": undefined,
                },
              },
            },
            "webAuth": [Circular],
            "webMessageHandler": WebMessageHandler {
              "warn": Warn {
                "disableWarnings": undefined,
              },
              "webAuth": [Circular],
            },
          },
          "warn": Warn {
            "disableWarnings": false,
          },
          "webAuth": [Circular],
        },
        "ssodataStorage": SSODataStorage {
          "storage": Storage {
            "handler": StorageHandler {
              "storage": CookieStorage {},
              "warn": Warn {
                "disableWarnings": undefined,
              },
            },
          },
        },
        "transactionManager": TransactionManager {
          "keyLength": 32,
          "namespace": "com.auth0.auth.",
          "options": Object {
            "_sendTelemetry": true,
            "_telemetryInfo": Object {
              "env": Object {
                "auth0.js": "9.13.2",
              },
              "name": "lock.js",
              "version": "11.24.1",
            },
            "_timesToRetryFailedRequests": 0,
            "audience": undefined,
            "clientID": "client-id",
            "domain": "test.com",
            "jwksURI": null,
            "leeway": 60,
            "nonce": undefined,
            "overrides": null,
            "plugins": PluginHandler {
              "plugins": Array [
                CordovaPlugin {
                  "extensibilityPoints": Array [
                    "popup.authorize",
                    "popup.getPopupHandler",
                  ],
                  "version": "9.13.2",
                  "webAuth": [Circular],
                },
              ],
            },
            "popupOrigin": undefined,
            "redirectUri": undefined,
            "responseMode": undefined,
            "responseType": undefined,
            "rootUrl": "https://test.com",
            "scope": undefined,
            "state": undefined,
            "tenant": "test",
            "token_issuer": "https://test.com/",
            "universalLoginPage": true,
          },
          "storage": Storage {
            "handler": StorageHandler {
              "storage": CookieStorage {},
              "warn": Warn {
                "disableWarnings": undefined,
              },
            },
          },
        },
        "webMessageHandler": WebMessageHandler {
          "warn": Warn {
            "disableWarnings": undefined,
          },
          "webAuth": [Circular],
        },
      },
      "domain": "test.com",
      "isUniversalLogin": false,
      "lockID": "lock-id",
    },
  },
}
`;
